### PR TITLE
Canonicalize targets of clean command.

### PR DESCRIPTION
**Pull request by fjhenigman**
This is impeding our ability on the ANGLE project to use GN with Visual Studio:
https://bugs.chromium.org/p/chromium/issues/detail?id=791295
Thanks for any suggestions.

### CODE REVIEW

**Comment by null77** [comment]
@nico @sgraham PTAL

**Comment by null77** [comment]
@atetubou are you an owner who would be able to review/land this?

File: src/clean.cc [hunk 1]
```diff
@@ -180,15 +180,21 @@ int Cleaner::CleanTargets(int target_count, char* targets[]) {
...
-    if (target) {
-      if (IsVerbose())
-        printf("Target %s\n", target_name);
-      DoCleanTarget(target);
-    } else {
-      Error("unknown target '%s'", target_name);
+    string target_name = targets[i];
+    uint64_t slash_bits;
+    string err;
+    if (!CanonicalizePath(&target_name, &slash_bits, &err)) {
```
Comment (atetubou) [inline src/clean.cc]: Can I ask you to show err if it fails to canonicalize?
```
Error("Failed to canonicalize '%s': %s", target_name,c_str(), err.c_str());
```

**Comment by atetubou** [comment]
Sorry, I'm developer interested in this project, but I'm not owner.

**Comment by null77** [comment]
@evmar PTAL at this CL, it's been waiting for an owner to review it for a month.

**Comment by sgraham** [comment]
Sorry, I went on vacation on Dec 20 so I just missed your ping, and am just getting back. I think this is probably fine, but as mentioned offline, it might be more effective for you to fix this in the GN generator as the turnaround time for a new released ninja version will be pretty long.

File: src/clean.cc (hunk 1 above)
Comment (nico) [reply src/clean.cc]: This doesn't look addressed.

Comment (fjhenigman) [reply src/clean.cc]: done

**Comment by nico** [comment]
Thanks!

**Pull request merged**